ci/azure-test-stable.yml

```diff
@@ -48,14 +67,25 @@ jobs:
       env:
         LOOM_MAX_PREEMPTIONS: 2
         CI: 'True'
-      displayName: ${{ crate.key }} - cargo test
-      workingDirectory: $(Build.SourcesDirectory)/${{ crate.key }}
-
-    # Run with each specified feature
-    - ${{ each feature in crate.value }}:
-      - script: cargo test --no-default-features --features ${{ feature }}
-        env:
-          LOOM_MAX_PREEMPTIONS: 2
-          CI: 'True'
-        displayName: ${{ crate.key }} - cargo test --features ${{ feature }}
-        workingDirectory: $(Build.SourcesDirectory)/${{ crate.key }}
+      displayName: ${{ crate }} - cargo test
+      workingDirectory: $(Build.SourcesDirectory)/${{ crate }}
+
+    # Run with all crate features
+    - script: cargo test --all-features
+      env:
+        LOOM_MAX_PREEMPTIONS: 2
+        CI: 'True'
+      displayName: ${{ crate }} - cargo test --all-features
+      workingDirectory: $(Build.SourcesDirectory)/${{ crate }}
+
+    # Check each specified feature works properly
+    # * --each-feature - run for each feature which includes --no-default-features and default features of package
+    # * --no-dev-deps - build without dev-dependencies to avoid https://github.com/rust-lang/cargo/issues/4866
+    - script: cargo hack check --each-feature --no-dev-deps
+      env:
+        LOOM_MAX_PREEMPTIONS: 2
+        CI: 'True'
+      displayName: ${{ crate }} - cargo hack check --each-feature
+      workingDirectory: $(Build.SourcesDirectory)/${{ crate }}
+      # FIXME(taiki-e): many features of current cargo-hack are not available on windows: https://github.com/taiki-e/cargo-hack/issues/3
+      condition: not(eq(variables['Agent.OS'], 'Windows_NT'))
```

Comment:
Unfortunately, many features of current cargo-hack are not available on windows. (https://github.com/taiki-e/cargo-hack/issues/3)

Reply:
However, it seems many features are not directly related to OS-specific dependencies (https://github.com/tokio-rs/tokio/pull/1695/commits/7aee1f85f691f7e125f11e6f7cca4b2b1b04a074), so for now, I think this is almost all right.

Reply:
I tried to fix this in https://github.com/taiki-e/cargo-hack/pull/6, but I couldn't fix it completely.